### PR TITLE
Allow sign in using personal accounts

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request is a combination of multiple previous prs that are about allowing private accounts:

- #3752 
- #3855 
- #3859
- #3806

- [x] Tests were added

Closes https://github.com/dodona-edu/dodona/issues/2597 .

### CODE REVIEW

**Comment by jorg-vr** [comment]
Tested everything together on naos and it works

File: config/locales/views/auth/nl.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ nl:
       higher-education: "Hoger onderwijs"
       other-education: "Andere onderwijsinstellingen"
       other: "Andere organisaties"
+      private-accounts: Private accounts
+      private-accounts-help: Gebruik deze login methodes als je niet bij een onderwijsinstelling aangesloten bent
```
Comment (bmesuere) [inline config/locales/views/auth/nl.yml:14]:
```suggestion
      private-accounts-help: Gebruik deze loginmethodes om je persoonlijke Google of Microsoft account te gebruiken als je niet bij een onderwijsinstelling bent aangesloten.
```

File: config/locales/views/auth/en.yml [hunk 2]
```diff
@@ -10,6 +10,8 @@ en:
       higher-education: "Higher education"
       other-education: "Other educational institutions"
       other: "Other institutions"
+      private-accounts: Private accounts
+      private-accounts-help: Use these sign in methods if you are associated with any educational institution
```
Comment (bmesuere) [inline config/locales/views/auth/en.yml:14]:
```suggestion
      private-accounts-help: Use these sign in methods to use your personal Google or Microsoft account if you are not associated with any educational institution.
```

File: config/locales/views/courses/en.yml [hunk 3]
```diff
@@ -17,6 +17,7 @@ en:
       registration-open_for_all-help_html: Anyone can register for this course.
       registration-open_for_institution-help_html: Only members of <span class="fill-institution">the configured institution</span> can register for this course.
       registration-closed-help_html: All new registrations are disabled.
+      registration-open_for_institutional_users-help_html: Anyone that belongs to an institution can register for this course. Users that signed in using a <b>private account</b>, such as a personal Gmail or Outlook account, <b>cannot</b> register.
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:20]:
```suggestion
      registration-open_for_institutional_users-help_html: Anyone that belongs to an institution can register for this course. Users that signed in using a <b>private account</b>, such as a personal Google or Microsoft account, <b>cannot</b> register.
```

File: config/locales/views/courses/en.yml [hunk 4]
```diff
@@ -98,6 +100,7 @@ en:
       registration-open_for_all-info: "Registrations are open for everyone: all users can register for this course."
       registration-open_for_institution-info: "Registrations are open for members of %{institution}: only they can register for this course."
       registration-closed-info: "Registrations are closed: users can't register until you change the registration settings. Users who are currently registered still have access."
+      registration-open_for_institutional_users-info: "Registrations are open for users that belong to any institution. Registrations are closed for users who signed in using a private account, such as a personal Gmail or Outlook account."
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:103]:
```suggestion
      registration-open_for_institutional_users-info: "Registrations are open for users that belong to any institution. Registrations are closed for users who signed in using a private account, such as a personal Google or Microsoft account."
```

File: config/locales/views/courses/nl.yml [hunk 5]
```diff
@@ -17,6 +17,7 @@ nl:
       registration-open_for_all-help_html: Iedereen kan zich voor deze cursus registreren.
       registration-open_for_institution-help_html: Enkel gebruikers van <span class="fill-institution">de ingestelde onderwijsinstelling</span> kunnen zich voor deze cursus registreren.
       registration-closed-help_html: Er zijn geen nieuwe registraties mogelijk.
+      registration-open_for_institutional_users-help_html: Iedereen die tot een instelling behoort kan zich voor deze cursus registreren. Gebruikers die zijn ingelogd met een <b>privéaccount</b>, zoals een persoonlijk Gmail- of Outlook-account, kunnen zich <b>niet</b> registreren.
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:20]:
```suggestion
      registration-open_for_institutional_users-help_html: Iedereen die tot een instelling behoort kan zich voor deze cursus registreren. Gebruikers die zijn ingelogd met een <b>privéaccount</b>, zoals een persoonlijk Google- of Microsoft-account, kunnen zich <b>niet</b> registreren.
```

File: config/locales/views/courses/nl.yml [hunk 6]
```diff
@@ -107,6 +109,7 @@ nl:
       registration-open_for_all-info: 'De registraties staan open voor iedereen: gebruikers die op de registratie-knop klikken worden geregistreerd.'
       registration-open_for_institution-info: 'De registraties staan open voor gebruikers van %{institution}: enkel zij kunnen zich registreren voor deze cursus.'
       registration-closed-info: 'De registraties zijn gesloten: niemand kan zich nog registreren tot de registratie-instelling wordt aangepast. Gebruikers die geregistreerd zijn behouden hun toegang.'
+      registration-open_for_institutional_users-info: 'De registraties staan open voor gebruikers die tot een instelling behoren. De registraties zijn gesloten voor gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Gmail- of Outlook-account.'
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:112]:
```suggestion
      registration-open_for_institutional_users-info: 'De registraties staan open voor gebruikers die tot een instelling behoren. De registraties zijn gesloten voor gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Google- of Microsoft-account.'
```

**Review by bmesuere — CHANGES_REQUESTED** [review]

**Review by bmesuere — APPROVED** [review]

**Pull request merged**